waggle-dance-api/src/main/java/com/hotels/bdp/waggledance/api/model/MetastoreTunnel.java

```diff
@@ -15,82 +15,6 @@
  */
 package com.hotels.bdp.waggledance.api.model;
 
-import javax.validation.constraints.Max;
-import javax.validation.constraints.Min;
-import javax.validation.constraints.Pattern;
+import com.hotels.hcommon.ssh.SshSettingsMapper;
 
-import org.hibernate.validator.constraints.NotBlank;
-
-import com.hotels.bdp.waggledance.api.validation.constraint.TunnelRoute;
-
-public class MetastoreTunnel {
-
-  private static final int DEFAULT_PORT = 22;
-  private static final String DEFAULT_LOCALHOST = "localhost";
-  private static final int DEFAULT_TIMEOUT_MILLIS = 60000; // 1 minute
-
-  private @NotBlank @TunnelRoute String route;
-  private @Min(1) @Max(65535) int port = DEFAULT_PORT;
-  private String localhost = DEFAULT_LOCALHOST;
-  private @NotBlank String privateKeys;
-  private @NotBlank String knownHosts;
-  private @Min(0) int timeout = DEFAULT_TIMEOUT_MILLIS;
-  @Pattern(regexp = "^(?i:yes\\b|no\\b)", message = "StrictHostKeyChecking can be set to 'yes' or 'no'")
-  private String strictHostKeyChecking = "yes";
-
-  public String getRoute() {
-    return route;
-  }
-
-  public void setRoute(String route) {
-    this.route = route;
-  }
-
-  public int getPort() {
-    return port;
-  }
-
-  public void setPort(int port) {
-    this.port = port;
-  }
-
-  public String getLocalhost() {
-    return localhost;
-  }
-
-  public void setLocalhost(String localhost) {
-    this.localhost = localhost;
-  }
-
-  public String getPrivateKeys() {
-    return privateKeys;
-  }
-
-  public void setPrivateKeys(String privateKeys) {
-    this.privateKeys = privateKeys;
-  }
-
-  public String getKnownHosts() {
-    return knownHosts;
-  }
-
-  public void setKnownHosts(String knownHosts) {
-    this.knownHosts = knownHosts;
-  }
-
-  public int getTimeout() {
-    return timeout;
-  }
-
-  public void setTimeout(int timeout) {
-    this.timeout = timeout;
-  }
-
-  public String getStrictHostKeyChecking() {
-    return strictHostKeyChecking;
-  }
-
-  public void setStrictHostKeyChecking(String strictHostKeyChecking) {
-    this.strictHostKeyChecking = strictHostKeyChecking;
-  }
-}
+public class MetastoreTunnel extends SshSettingsMapper {}
```

Comment:
Nice, look at all that repeated code gone!

Reply:
Well, we don't want to mention the Metastore in hcommon-ssh as it's supposed to only know about ssh, not metastores. So the compromise was to keep this here so we have a descriptive and backwards compatible name for the YAML/Spring Boot properties but at least all the repeated code is gone.